rest/src/main/kotlin/builder/components/CompositeComponentBuilder.kt

```diff
@@ -0,0 +1,34 @@
+package dev.kord.rest.builder.components
+
+import dev.kord.common.annotation.KordPreview
+import dev.kord.common.entity.ComponentType
+import dev.kord.common.entity.DiscordComponent
+import dev.kord.common.entity.optional.optional
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+@KordPreview
+class CompositeComponentBuilder {
+    var components: MutableList<DiscordComponent> = mutableListOf()
+
+    @OptIn(ExperimentalContracts::class)
+    inline fun actionRow(builder: CompositeComponentBuilder.() -> Unit) {
+        contract {
+            callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
+        }
+
+        components.add(ActionRow(builder))
+    }
```

Comment:
It does not make sense for this to be a `CompositeComponentBuilder`, the discord docs state that nesting action rows inside action rows is forbidden. <https://discord.com/developers/docs/interactions/message-components#actionrow>